src/Illuminate/Database/Eloquent/Relations/Concerns/InteractsWithPivotTable.php

```diff
@@ -67,7 +67,7 @@ public function toggle($ids, $touch = true)
      * Sync the intermediate tables with a list of IDs without detaching.
      *
      * @param  \Illuminate\Support\Collection|\Illuminate\Database\Eloquent\Model|array  $ids
-     * @return array
+     * @return array{attached: array, detached: array, updated: array}
```

Comment:
All of the `array` inner types should really be defined (e.g., `array<string, mixed>` for instance)

Reply:
I agree, but I don't think we can be more specific than `array<array-key, mixed>`, which is the same as `array`.